src_Core/CPU/Core.bsv

```diff
@@ -1165,49 +1149,42 @@ module mkCore#(CoreId coreId)(Core);
      // Each cache and TLB pair uses the same struct (EventsCache), but the cache accesses
      // different fields than the TLB, which makes it safe to combine them
 
-     Reg#(EventsCache) events_llc_reg <- mkRegU;
-     Reg#(Vector#(7, Bit#(1))) events_tgc_reg <- mkRegU;
+     Reg#(EventsLL) events_llc_reg <- mkRegU;
+     Reg#(EventsCacheCore) events_tgc_reg <- mkRegU;
      rule report_events;
          EventsCore events = unpack(pack(commitStage.events));
          events.evt_REDIRECT = zeroExtend(pack(fetchStage.redirect_evt));
          hpm_core_events[1] <= events;
      endrule
 
-     Vector #(1, Bit #(Report_Width)) null_evt = replicate (0);
-     Vector #(31, Bit #(Report_Width)) mem_core_evts_vec =  to_large_vector (coreFix.memExeIfc.events);
-     Vector #(31, Bit #(Report_Width)) other_core_evts_vec = to_large_vector (hpm_core_events[0]);
-     Vector #(31, Bit #(Report_Width)) core_evts_vec = unpack(pack(mem_core_evts_vec) | pack(other_core_evts_vec));
-     EventsCache instMem = unpack(pack(iMem.events) | pack(iTlb.events));
-     Vector #(16, Bit #(Report_Width)) imem_evts_vec = to_large_vector (instMem);
-     EventsCache dataMem = unpack(pack(dMem.events) | pack(dTlb.events));
-     Vector #(16, Bit #(Report_Width)) dmem_evts_vec = to_large_vector (dataMem);
-     Vector #(32, Bit #(Report_Width)) tgc_evts_vec = to_large_vector (events_tgc_reg);
-     EventsCache llMem = unpack(pack(events_llc_reg) | pack(l2Tlb.events));
-     Vector #(16, Bit #(Report_Width)) llc_evts_vec = to_large_vector (llMem);
+     EventsCore core_evts = unpack(pack(coreFix.memExeIfc.events) | pack(hpm_core_events[0]));
+     EventsL1I imem_evts = unpack(pack(iMem.events) | pack(iTlb.events));
+     EventsL1D dmem_evts = unpack(pack(dMem.events) | pack(dTlb.events));
+     EventsCacheCore tgc_evts = events_tgc_reg;
+     EventsLL llmem_evts = unpack(pack(events_llc_reg) | pack(l2Tlb.events));
+     Maybe#(EventsTransExe) mab_trans_exe = tagged Invalid;
+
```

Comment:
Now that we have a structured set of events that isn't "just a bitfield", I am wondering if combining them with a `pack` + `|` is the most idiomatic way to go. Not sure I have a fundamentally better idea for it, really you'd want to use a `Monoid` `mappend` I think, but there is no `Monoid` class in Bluespec...